src/main/ngapp/accio/src/app/mask.directive.ts

```diff
@@ -0,0 +1,140 @@
+
+    // tolerance will come from user input
+    const tolerance = 20;
+
+    // returns an array indicies of each pixel in the mask
+    const maskPixels = this.magicWandService.floodfill(this.originalData, Math.floor(xCoord / this.scale), Math.floor(yCoord / this.scale), tolerance);
+
+    // alpha value hardcoded for now
+    this.drawMask(maskPixels, imgData, 1);
+  }
+
+  /**
+   *  Creates a UI mask to the user by reading in all values in the given maskPixels
+   *  Clears the canvas fisrt so the mask can be saved as a new image and then over-
+   *    lays the mask on top of the scaled image. 
+   *  Adds all pixels to mask by giving alpha value of 255, changing global alpha on 
+   *    maskImage will multiply 255 by globalAlpha = [0,1]
+   *  maskPixels: set of coordinates indexed by red value of pixels to be in mask
+   *  imgData: Uint8clampedArray of all pixel data in scaled image. 
+   *  alphaValue: the number the user passes in on a slide bar for transparency of the mask
+   */
+  private drawMask(maskPixels: Set<number>, imgData: ImageData, alphaValue: number): void {
+    //clear old context
+    this.ctx.clearRect(0,0,this.canvas.nativeElement.width, this.canvas.nativeElement.height);
+    
+    //new imagedata creates transparent black rectangle
+    let img = new ImageData(imgData.width, imgData.height);
+
+    // access all pixels in original mask, scale them up to image on UI width
+    // ***** Need flood fill to pass back total mask, not just new one created, used for undo/redo as well
+    for (let pixel of maskPixels) {
+      let y = this.getY(this.originalData.width, pixel);
+      let x = this.getX(this.originalData.width, pixel, y);
+      img.data[this.getPixel(x, y, imgData.width) + 3] = 255;
+    } 
+    // Put the new img data on ctx and save as an img
+    this.ctx.putImageData(img, 0, 0);
+    
+    const mask = this.canvas.nativeElement.toDataURL();
+
+    // clear canvas again to prepare for image and mask
+    this.ctx.clearRect(0,0,this.canvas.nativeElement.width, this.canvas.nativeElement.height);
+  
+    console.log('adding mask'); 
+    this.drawImageAndMask(imgData, mask, alphaValue)
+    console.log('new mask added');
+  }
+
+  /**
+  * Changes the alpha value of the mask displayed 
+  *  alphaValue: the number the user passes in on a slide bar for transparency of the mask
+  */
+  private changeAlpha(alphaValue: number): void {
+    if (alphaValue == 0) {
+      alphaValue = .01;
+    }
+    this.ctx.globalAlpha = alphaValue;
+    var newMask = new Image();
+    newMask.src = this.maskImage.src;
+    newMask.onload = () => {
+        this.ctx.drawImage(newMask, 0, 0);
+    };
+  }
+
+  /**
+  * draws the original image and updated mask on context
+  * imgData: ImageData from the original image
+  * mask: source of new mask 
+  * alphaValue: the number the user passes in on a slide bar for transparency of the mask
```

Comment:
Same as prev comment: You can use the '@' symbol as a prefix to denote parameters